report_wkhtmltopdf_param/tests/test_report_paperformat.py

```diff
@@ -23,6 +36,8 @@ def test_wkhtmltopdf_valid_parameter(self):
                 report_paperformat.update(
                     {"custom_params": [(0, 0, {"name": "--disable-smart-shrinking"})]}
                 )
+            except ParserError:
+                _logger.error("Document is empty")
```

Comment:
Hey this is a change to another module, did this come from you in order to make it green? Or from elsewhere

Reply:
Yeah did this to make it green.

Reply:
Ok but then in which line does it throw ParserError when you run the tests?

https://github.com/OCA/reporting-engine/blob/15.0/report_wkhtmltopdf_param/models/report_paperformat.py#L41

Reply:
so I was due to a _**"Document is Empty"**_ error this can be seen at [line 306](https://github.com/OCA/reporting-engine/runs/5760772752?check_suite_focus=true#step:8:306) from the previous build, that is  why I added the exception part. Mostly on a new demo db.

Reply:
Actually, something else was going on. In the [15.0 migration](https://github.com/OCA/reporting-engine/pull/579/files) in [this commit](https://github.com/OCA/reporting-engine/commit/3638d7d0b5510da400334bfa9b0ab18a3f615197) the `content_html` array was replaced by `sample_html`, and no-one complained about it during review, maybe it was overlooked.

As a result, `_run_wkhtmltopdf` was run directly with a HTML string as the second parameter, instead of a `bodies` array of strings like expected.

So `for i, body in enumerate(bodies)` splits up the string into its 293 characters which miraculously does not lead to problems in the normal test, but when we added our code, we are overriding `_run_wkhtmltopdf` and actually trying to decode each of the bodies from HTML to a structure, and then it gives an error.

The correct solution would thus be not to change the test to deal with ParserError, but to prevent ParserError in the first place by reverting the change that was made in the module.